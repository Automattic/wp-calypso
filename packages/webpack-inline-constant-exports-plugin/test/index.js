/**
 * External dependencies
 */
const fs = require( 'fs' );
const path = require( 'path' );
const rimraf = require( 'rimraf' );
const webpack = require( 'webpack' );
const InlineConstantExportsPlugin = require( '..' );

describe( 'webpack-inline-constant-exports-plugin', () => {
	const fixturesDirectory = path.join( __dirname, 'fixtures' );
	const buildDirectory = path.join( __dirname, 'build' );

	beforeAll( () => {
		rimraf.sync( buildDirectory );
	} );

	afterAll( () => {
		rimraf.sync( buildDirectory );
	} );

	test( 'should produce expected output', done => {
		const inputDirectory = path.join( fixturesDirectory, 'basic' );
		const outputDirectory = path.join( buildDirectory, 'basic' );
		const config = {
			context: inputDirectory,
			entry: './index.js',
			mode: 'production',
			optimization: {
				runtimeChunk: true,
				moduleIds: 'named',
				chunkIds: 'named',
				minimize: false,
			},
			output: {
				path: outputDirectory,
			},
			plugins: [
				new InlineConstantExportsPlugin( [ /\/actions\.js$/, /\/plans\.js$/, /\/constants\.js$/ ] ),
			],
		};

		webpack( config, err => {
			expect( err ).toBeNull();

			const outputFile = path.join( outputDirectory, 'main.js' );
			const outputFileContent = fs.readFileSync( outputFile, 'utf8' );
			expect( outputFileContent ).toMatchSnapshot( 'Output bundle should match snapshot' );

			done();
		} );
	} );
} );
