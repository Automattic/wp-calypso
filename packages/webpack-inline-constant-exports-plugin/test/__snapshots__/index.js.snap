// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-inline-constant-exports-plugin should produce expected output: Output bundle should match snapshot 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"main\\"],{

/***/ \\"./index.js\\":
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);

// CONCATENATED MODULE: ./plans.js
/*
 * Export two plan constants and a a constant array of all plans. The array should not be inlined
 * and the module should stay in the dependency graph.
 */
const BLOGGER = 'BLOGGER_PLAN';
const PREMIUM = 'PREMIUM_PLAN';
/* harmony default export */ var plans = ([ BLOGGER, PREMIUM ]);

// CONCATENATED MODULE: ./paths.js
/*
 * The export is eligible for inlining, but the module is not specified in the plugin config.
 * Therefore no inlining should happen.
 */
const HOME_PATH = '/';

// CONCATENATED MODULE: ./index.js
/**
 * Internal dependencies
 */





/* eslint-disable no-console */
console.log( /* inline */ 'PLANS_REQUEST', /* inline */ 'PLANS_RECEIVE' );
console.log( /* inline */ 'BLOGGER_PLAN', /* inline */ 'PREMIUM_PLAN', plans );
console.log( /* inline */ 42, /* inline */ 3.14159, /* inline */ true, /* inline */ false, /* inline */ null );
console.log( HOME_PATH );


/***/ })

},[[\\"./index.js\\",\\"runtime~main\\"]]]);"
`;
